src/main/java/nourl/mythicmetals/mixin/AbstractMinecartEntityTypeMixin.java

```diff
@@ -0,0 +1,36 @@
+package nourl.mythicmetals.mixin;
+
+import net.minecraft.entity.vehicle.AbstractMinecartEntity;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+@Mixin(AbstractMinecartEntity.Type.class)
+public abstract class AbstractMinecartEntityTypeMixin {
+    @Shadow
+    @Final
+    @Mutable
+    private static AbstractMinecartEntity.Type[] field_7673;
+
+    private static final AbstractMinecartEntity.Type BANGLUM_TNT = mythicmetals$addType("BANGLUM_TNT");
```

Comment:
This should be annoted with `@Unique`

Reply:
Should it? That would only take effect if a constant named `BANGLUM_TNT` were already present in the `AbstractMinecartEntity$Type` enum, in which case the call to `Enum#valueOf` in `MythicMetals` would result in undefined behavior, since the `BANGLUM_TNT` constant we're looking for (the one added by Mythic Metals' mixin) would be named something else by the mixin processor.

Then again, *without* `@Unique`, it'll just overwrite whatever `BANGLUM_TNT` constant may already exist, which could also be considered "undefined behavior"... so the annotation doesn't really matter either way, does it?

Reply:
That is true. I made the suggestion as it is warned about with McDev. It is very unlikely this field would exist in other contexts.
I think its fine to leave as is for now

Reply:
That's funny; McDev didn't warn me about it :P